pkg/kp/rcstore/rc.go

```diff
@@ -0,0 +1,95 @@
+package rcstore
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+)
+
+type ReplicationControllerDesires struct {
+	replicasDesired int
+	disabled        bool
+}
+
+type ReplicationControllerData struct {
+	// It is expected that these fields will never be changed after an RC is created.
+	id           string
+	manifest     *pods.Manifest
+	nodeSelector labels.Selector
+	podLabels    labels.Set
+
+	watcher desireWatcher
+
+	// It is expected that these fields can be changed.
+	ReplicationControllerDesires
+}
+
+// ReplicationControllerStore represents an interface for persisting replication controllers to Consul,
+// as well as restoring replication controllers from Consul.
+type ReplicationControllerStore interface {
+	// Create creates a replication controller with the specified manifest and selectors.
+	// The node selector is used to determine what nodes the replication controller may schedule on.
+	// The pod label set is:
+	// 1) applied to every pod the replication controller schedules,
+	// 2) used (in addition to replication controller ID) to determine what pods are already scheduled.
+	Create(manifest pods.Manifest, nodeSelector labels.Selector, podLabels labels.Set) (ReplicationRequest, error)
+
+	// List lists all of the replication controllers.
+	List() ([]ReplicationRequest, error)
+}
+
+type desireWatcher interface {
```

Comment:
It could very well be `intent`.  we'd have to be careful not to confuse replication controller intent with pod intent though. Is the following sentence confusing to you?

"Replication controllers try to bring their reality in line with their intent by scheduling manifests into the intent store"